nixos/modules/services/databases/postgrest.nix

```diff
@@ -0,0 +1,258 @@
+            type = with lib.types; nullOr str;
+            default = "127.0.0.1";
+            description = ''
+              Where to bind the PostgREST web server.
+
+              ::: {.note}
+              The admin server will also bind here, but potentially exposes sensitive information.
+              Make sure you turn off the admin server, when opening this to the public.
+
+              <https://github.com/PostgREST/postgrest/issues/3956>
+              :::
+            '';
+          };
+
+          server-port = lib.mkOption {
+            type = with lib.types; nullOr port;
+            default = null;
+            example = "3000";
+            description = ''
+              The TCP port to bind the web server.
+            '';
+          };
+
+          server-unix-socket = lib.mkOption {
+            type = with lib.types; nullOr path;
+            default = "/run/postgrest/postgrest.sock";
+            description = ''
+              Unix domain socket where to bind the PostgREST web server.
+            '';
+          };
+        };
+      };
+      default = { };
+      description = ''
+        PostgREST configuration as documented in:
+        <https://docs.postgrest.org/en/stable/references/configuration.html#list-of-parameters>
+
+        ::: {.note}
+        The `settings.jwt-secret` option is blocked.
+        Use `jwtSecretFile` instead.
+        :::
+      '';
+      example = lib.literalExpression ''
+        {
+          db-anon-role = "anon";
+          db-uri = "dbname=postgres";
+          "app.settings.custom" = "value";
+        }
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = (cfg.settings.server-port == null) != (cfg.settings.server-unix-socket == null);
+        message = ''
+          PostgREST can listen either on a TCP port or on a unix socket, but not both.
+          Please set one of `settings.server-port` or `settings.server-unix-socket` to `null`.
+
+          <https://docs.postgrest.org/en/stable/references/configuration.html#server-unix-socket>
+        '';
+      }
+    ];
+
+    warnings =
+      if cfg.settings.admin-server-port != null && cfg.settings.server-host != "127.0.0.1" then
+        [
+          "The PostgREST admin server is potentially listening on a public host. This may expose sensitive information via the `/config` endpoint."
+        ]
+      else
+        [ ];
+
+    systemd.services.postgrest = {
+      description = "PostgREST";
+
+      wantedBy = [ "multi-user.target" ];
+      after = [
+        "network.target"
+        "postgresql.service"
+      ];
+
+      serviceConfig = {
+        DynamicUser = true;
+        LoadCredential =
+          lib.optional (cfg.dbPasswordFile != null) "db_password:${cfg.dbPasswordFile}"
+          ++ lib.optional (cfg.jwtSecretFile != null) "jwt_secret:${cfg.jwtSecretFile}";
+        RuntimeDirectory = "postgrest";
+        User = "postgrest";
+      };
```

Comment:
How about adding:
```
Restart = "always";
RestartSec = "10s";
```

Nginx has the same config:

https://github.com/NixOS/nixpkgs/blob/882031e265e22b579d25a3be1981d8edd604a3f5/nixos/modules/services/web-servers/nginx/default.nix#L1252-L1253

Less time could be good too, like `3s`.